Sources/CCryptoBoringSSL/CMakeLists.txt

```diff
@@ -0,0 +1,345 @@
+  crypto/x509/by_dir.c
+  crypto/x509/by_file.c
+  crypto/x509/i2d_pr.c
+  crypto/x509/rsa_pss.c
+  crypto/x509/t_crl.c
+  crypto/x509/t_req.c
+  crypto/x509/t_x509.c
+  crypto/x509/t_x509a.c
+  crypto/x509/x509.c
+  crypto/x509/x509_att.c
+  crypto/x509/x509_cmp.c
+  crypto/x509/x509_d2.c
+  crypto/x509/x509_def.c
+  crypto/x509/x509_ext.c
+  crypto/x509/x509_lu.c
+  crypto/x509/x509_obj.c
+  crypto/x509/x509_r2x.c
+  crypto/x509/x509_req.c
+  crypto/x509/x509_set.c
+  crypto/x509/x509_trs.c
+  crypto/x509/x509_txt.c
+  crypto/x509/x509_v3.c
+  crypto/x509/x509_vfy.c
+  crypto/x509/x509_vpm.c
+  crypto/x509/x509cset.c
+  crypto/x509/x509name.c
+  crypto/x509/x509rset.c
+  crypto/x509/x509spki.c
+  crypto/x509/x_algor.c
+  crypto/x509/x_all.c
+  crypto/x509/x_attrib.c
+  crypto/x509/x_crl.c
+  crypto/x509/x_exten.c
+  crypto/x509/x_info.c
+  crypto/x509/x_name.c
+  crypto/x509/x_pkey.c
+  crypto/x509/x_pubkey.c
+  crypto/x509/x_req.c
+  crypto/x509/x_sig.c
+  crypto/x509/x_spki.c
+  crypto/x509/x_val.c
+  crypto/x509/x_x509.c
+  crypto/x509/x_x509a.c
+  crypto/x509v3/pcy_cache.c
+  crypto/x509v3/pcy_data.c
+  crypto/x509v3/pcy_lib.c
+  crypto/x509v3/pcy_map.c
+  crypto/x509v3/pcy_node.c
+  crypto/x509v3/pcy_tree.c
+  crypto/x509v3/v3_akey.c
+  crypto/x509v3/v3_akeya.c
+  crypto/x509v3/v3_alt.c
+  crypto/x509v3/v3_bcons.c
+  crypto/x509v3/v3_bitst.c
+  crypto/x509v3/v3_conf.c
+  crypto/x509v3/v3_cpols.c
+  crypto/x509v3/v3_crld.c
+  crypto/x509v3/v3_enum.c
+  crypto/x509v3/v3_extku.c
+  crypto/x509v3/v3_genn.c
+  crypto/x509v3/v3_ia5.c
+  crypto/x509v3/v3_info.c
+  crypto/x509v3/v3_int.c
+  crypto/x509v3/v3_lib.c
+  crypto/x509v3/v3_ncons.c
+  crypto/x509v3/v3_ocsp.c
+  crypto/x509v3/v3_pci.c
+  crypto/x509v3/v3_pcia.c
+  crypto/x509v3/v3_pcons.c
+  crypto/x509v3/v3_pmaps.c
+  crypto/x509v3/v3_prn.c
+  crypto/x509v3/v3_purp.c
+  crypto/x509v3/v3_skey.c
+  crypto/x509v3/v3_utl.c)
+if(APPLE)
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|amd64")
+    target_sources(CCryptoBoringSSL PRIVATE
+      crypto/chacha/chacha-x86_64.mac.x86_64.S
+      crypto/cipher_extra/aes128gcmsiv-x86_64.mac.x86_64.S
+      crypto/cipher_extra/chacha20_poly1305_x86_64.mac.x86_64.S
+      crypto/fipsmodule/aesni-gcm-x86_64.mac.x86_64.S
+      crypto/fipsmodule/aesni-x86_64.mac.x86_64.S
+      crypto/fipsmodule/ghash-ssse3-x86_64.mac.x86_64.S
+      crypto/fipsmodule/ghash-x86_64.mac.x86_64.S
+      crypto/fipsmodule/md5-x86_64.mac.x86_64.S
+      crypto/fipsmodule/p256-x86_64-asm.mac.x86_64.S
+      crypto/fipsmodule/p256_beeu-x86_64-asm.mac.x86_64.S
+      crypto/fipsmodule/rdrand-x86_64.mac.x86_64.S
+      crypto/fipsmodule/rsaz-avx2.mac.x86_64.S
+      crypto/fipsmodule/sha1-x86_64.mac.x86_64.S
+      crypto/fipsmodule/sha256-x86_64.mac.x86_64.S
+      crypto/fipsmodule/sha512-x86_64.mac.x86_64.S
+      crypto/fipsmodule/vpaes-x86_64.mac.x86_64.S
+      crypto/fipsmodule/x86_64-mont.mac.x86_64.S
+      crypto/fipsmodule/x86_64-mont5.mac.x86_64.S)
+  endif()
+endif()
```

Comment:
We should cover windows and Linux as well, but that is okay to have as a follow up.

Reply:
Which ones are for Windows? I don't see any `*x86_64*` for it.

Reply:
Windows has some assembly routines for x86, not for x64.  I don't think that it matters too much if we don't get that variant correct.  I don't currently intend to ship a 32-bit toolchain.  For Windows x86, use of swift-crypto as a client should go through SPM, which will handle it properly.  If someone intends to ship a non-x64 variant of the toolchain, we should handle it at that time.

Reply:
OK. Added Linux https://github.com/apple/swift-crypto/pull/60/commits/6d2e5c1567a3bad8560c0478e370f35475d998da